SimPEG/EM/Analytics/FDEM_fields.py

```diff
@@ -0,0 +1,241 @@
+from __future__ import division
+import numpy as np
+from scipy.constants import mu_0, pi
+from scipy.special import erf
+from SimPEG import Utils
+
+def E_from_ElectricDipoleWholeSpace(XYZ, srcLoc, sig, f, current=1., length=1., orientation='X', mu=mu_0):
+    epsilon = 8.854187817*(10.**-12)
+    omega = 2.*np.pi*f
+    sig_hat = sig + 1j*omega*epsilon
+
+    XYZ = Utils.asArray_N_x_Dim(XYZ, 3)
+    # Check
+    if XYZ.shape[0] > 1 & f.shape[0] > 1:
+        raise Exception("I/O type error: For multiple field locations only a single frequency can be specified.")
+
+    dx = XYZ[:,0]-srcLoc[0]
```

Comment:
looks like this and the following 2 lines are used in a number of places, perhaps turn it into a small function?

```
def getSrcCenteredGrid(XYZ, srcLoc):
    # center the grid
    return XYZcentered
```